cmd/goplugin-aws/resource/instance_types.go

```diff
@@ -6,140 +6,140 @@ import (
-	InstanceLifecycle                 string            `puppet:"type=>String, kind=>given_or_derived"`
-	Platform                          string            `puppet:"type=>String, kind=>given_or_derived"`
-	PrivateDnsName                    string            `puppet:"type=>String, kind=>given_or_derived"`
-	ProductCodes                      []ProductCode     `puppet:"type=>Optional[Array[Aws::ProductCode]], value=>[]"`
-	PublicDnsName                     string            `puppet:"type=>String, kind=>given_or_derived"`
-	PublicIpAddress                   string            `puppet:"type=>String, kind=>given_or_derived"`
-	RamDiskId                         string            `puppet:"type=>String, kind=>given_or_derived"`
-	RootDeviceName                    string            `puppet:"type=>String, kind=>given_or_derived"`
-	RootDeviceType                    string            `puppet:"type=>String, kind=>given_or_derived"`
-	SecurityGroups                    []GroupIdentifier `puppet:"type=>Optional[Array[Aws::GroupIdentifier]], value=>[]"`
-	SourceDestCheck                   bool              `puppet:"type=>Boolean, kind=>given_or_derived"`
-	SpotInstanceRequestId             string            `puppet:"type=>String, kind=>given_or_derived"`
-	SriovNetSupport                   string            `puppet:"type=>String, kind=>given_or_derived"`
-	State                             InstanceState     `puppet:"type=>Optional[Aws::InstanceState], value=>undef"`
-	StateReason                       StateReason       `puppet:"type=>Optional[Aws::StateReason], value=>undef"`
-	StateTransitionReason             string            `puppet:"type=>String, kind=>given_or_derived"`
+	Monitoring                        *Monitoring       `puppet:"type=>Optional[Aws::Monitoring], value=>undef"`
+	Placement                         *Placement        `puppet:"type=>Optional[Aws::Placement], value=>undef"`
+	PrivateIpAddress                  string            `puppet:"type=>String, value=>''"`
+	RamdiskId                         string            `puppet:"type=>String, value=>''"`
+	SubnetId                          string            `puppet:"type=>String, value=>''"`
+	UserData                          string            `puppet:"type=>String, value=>''"`
+	OwnerId                           string            `puppet:"type=>String, value=>''"`
+	RequesterId                       string            `puppet:"type=>String, value=>''"`
+	ReservationId                     string            `puppet:"type=>String, value=>''"`
+	AmiLaunchIndex                    int64             `puppet:"type=>Integer, value=>0"`
+	Architecture                      string            `puppet:"type=>String, value=>''"`
+	EnaSupport                        bool              `puppet:"type=>Boolean, value=>false"`
+	Hypervisor                        string            `puppet:"type=>String, value=>''"`
+	InstanceId                        string            `puppet:"type=>String, value=>''"`
```

Comment:
This will not work as things stands today. The `given_or_derived` tells the WF-engine to exclude the field when comparing desired and actual state to determine if things changed. Just using a default value won't have that effect. Since the InstanceId is never present in the manifest (and hence never present in the desired state) the instance will always be considered in need for update. At present, such fields must be given_or_derived and pointers.

This is subject to change. I'm planning to add a `provided_attributes` field to the coming `Resource` annotation where such fields are listed by name in an array. I hope to have a PR for that by EOD today.